DevelopGitHub/GitHubExtension/GitHubExtension.Notes.WebApi/Controllers/NoteController.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Threading.Tasks;
+using System.Web.Http;
+using GitHubExtension.Notes.WebApi.Commands;
+using GitHubExtension.Notes.WebApi.Constants;
+using GitHubExtension.Notes.WebApi.Mappers;
+using GitHubExtension.Notes.WebApi.Queries;
+using GitHubExtension.Notes.WebApi.ViewModels;
+
+namespace GitHubExtension.Notes.WebApi.Controllers
+{
+    public class NoteController : ApiController
+    {
+        private INoteCommands commands;
+        private INoteQueries queries;
+
+        public NoteController(INoteCommands commands, INoteQueries queries)
+        {
+            this.commands = commands;
+            this.queries = queries;
+        }
+
+        [Route(RouteConstants.GetNoteRoute)]
+        public async Task<IHttpActionResult> GetNote([FromUri] int noteId)
+        {
+            var note = await queries.GetNote(noteId);
+            if (note == null)
+            {
+                return NotFound();
+            }
+            var noteModel = note.ToNoteViewModel();
+            if (!ModelState.IsValid)
```

Comment:
You are get note from DB a than verify that is model state is valid? I don't understand this logic.